mod/state-transition/pkg/core/state_processor_committee.go

```diff
@@ -36,8 +37,16 @@ func (sp *StateProcessor[
 		return nil, err
 	}
 
+	// filter out validators whose effective balance is not sufficient to validate
+	activeVals := make([]ValidatorT, 0, len(vals))
+	for _, val := range vals {
+		if val.GetEffectiveBalance() > math.U64(sp.cs.EjectionBalance()) {
+			activeVals = append(activeVals, val)
+		}
+	}
```

Comment:
you may ask why we didn't do it so far. The reason is so far:

- Validator are activated as soon as they are added
- They never reduces their stake

We need to change this to allow a capped validator set where validators may leave due to a validator with more stake coming in

Reply:
I think this is still incomplete.
Validators whose effective balance is below or equal to ejection should not only be marked as non-validator for consensus. Instead they should be actively removed and their balance eventually returned.
@calbera thoughts?